### PR TITLE
Wave limit game rule in custom maps

### PR DESCRIPTION
**Pull request by JniTrRny**
Unlocked the wave limit game rule to have it work in custom games.
image
0 to disable (wave goes indefinitely), 1 to make your game win instantly (why?)
If the wave has reached the limit (say wave 30/30) and the enemy units have been cleared, game over menu will pop up.

Also rearranged wave sending and wave timer so the rules look more clean when wave timer is disabled, and make everything disabled when waves rule is off.

NOTE: how would wave limit in PvP works? since the winning team will always be the `defaultTeam`.

---
If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by Darkness6030** [comment]
Players of my server voted for this feature:
Screenshot_2023-05-04-09-48-48-723_com discord-edit

File: core/src/mindustry/ui/dialogs/CustomRulesDialog.java [hunk 1]
```diff
@@ -280,7 +281,7 @@ void setup(){
 
             main.button("[#" + team.color +  "]" + team.localized() + (team.emoji.isEmpty() ? "" : "[] " + team.emoji), Icon.downOpen, Styles.togglet, () -> {
                 shown[0] = !shown[0];
-            }).marginLeft(14f).width(260f).height(55f).checked(a -> shown[0]).row();
+            }).marginLeft(14f).width(260f).height(55f).checked(a -> shown[0]).update(t -> ((Image)t.getChildren().get(1)).setDrawable(shown[0] ? Icon.upOpen : Icon.downOpen)).row();
```
Comment (Anuken) [inline core/src/mindustry/ui/dialogs/CustomRulesDialog.java:284]: checked() uses update(), and calling update() again overwrites the callback, so this breaks the checking

File: core/src/mindustry/ui/dialogs/CustomRulesDialog.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ private <T extends UnlockableContent> void showBanned(String title, ContentType
 
                 dialog.addCloseButton();
                 dialog.show();
-            }).size(300f, 64f);
+            }).size(300f, 64f).disabled(b -> set.size == content.<T>getBy(type).select(pred).size);
```
Comment (Anuken) [inline core/src/mindustry/ui/dialogs/CustomRulesDialog.java:104]: `select(..).count` allocates, use `count(...)`

File: core/assets/bundles/bundle.properties [hunk 3]
```diff
@@ -1237,6 +1237,7 @@ rules.buildcostmultiplier = Build Cost Multiplier
 rules.buildspeedmultiplier = Build Speed Multiplier
 rules.deconstructrefundmultiplier = Deconstruct Refund Multiplier
 rules.waitForWaveToEnd = Waves Wait for Enemies
+rules.wavelimit = Wave Limit
```
Comment (Anuken) [inline core/assets/bundles/bundle.properties:1240]: this doesn't make clear that the game ends when the target is hit - maybe call it "Map Ends After Wave"?

**Review by Anuken — CHANGES_REQUESTED** [review]

**Pull request merged**